packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,18 +37,112 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+// Examples can assume:
+// ScaffoldGeometry scaffoldGeometry;
+
+/// Geometry information for scaffold components.
+///
+/// To get a [ValueNotifier] for the scaffold geometry call
+/// [Scaffold.geometryOf].
+@immutable
+class ScaffoldGeometry {
+  const ScaffoldGeometry({
+    this.bottomNavigationBarTop,
+    this.floatingActionButtonArea,
+    this.floatingActionButtonScale: 1.0,
+  });
+
+  /// The distance from the scaffold's top edge to the top edge of the
+  /// rectangle in which the [Scaffold.bottomNavigationBar] bar is being laid
+  /// out.
+  ///
+  /// When there is no [Scaffold.bottomNavigationBar] set, this will be null.
+  final double bottomNavigationBarTop;
+
+  /// The rectangle in which the scaffold is laying out
+  /// [Scaffold.floatingActionButton].
+  ///
+  /// The floating action button might be scaled inside this rectangle, to get
+  /// the bounding rectangle in which the floating action is painted scale this
+  /// value by [floatingActionButtonScale].
+  ///
+  /// ## Sample code
+  ///
+  /// ```dart
+  ///  final Rect scaledFab = Rect.lerp(
+  ///    scaffoldGeometry.floatingActionButtonArea.center & Size.zero,
+  ///    scaffoldGeometry.floatingActionButtonArea,
+  ///    scaffoldGeometry.floatingActionButtonScale
+  ///  );
+  /// ```
+  ///
+  /// This is null when there is no floating action button showing.
+  final Rect floatingActionButtonArea;
+
+  /// The amount by which the [Scaffold.floatingActionButton] is scaled.
+  ///
+  /// To get the bounding rectangle in which the floating action button is
+  /// painted scaled [floatingActionPosition] by this proportion. 
+  ///
+  /// This will be 0 when there is no [Scaffold.floatingActionButton] set.
+  final double floatingActionButtonScale;
+}
+
+class _ScaffoldGeometryNotifier extends ValueNotifier<ScaffoldGeometry> {
+  _ScaffoldGeometryNotifier(ScaffoldGeometry geometry, this.context)
+    : assert (context != null),
+      super(geometry);
+
+  final BuildContext context;
+
+  @override
+  ScaffoldGeometry get value {
+    assert(() {
+      final RenderObject renderObject = context.findRenderObject();
+      if (renderObject == null || !renderObject.owner.debugDoingPaint)
+        throw new FlutterError(
+            'Scaffold.geometryOf() must only be accessed during the paint phase.\n\n'
```

Comment:
only one `\n` after the first line (see also `FlutterError.message` documentation)

Reply:
done